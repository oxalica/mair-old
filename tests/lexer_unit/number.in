1
1i32
1.2e3f64
1.2e-3
1e+3
0xEf64
0o__1_07f64
0b__1_01
1__3_.2_8_e_4_
1__3_.2_8_e-_4_
1__3_.2_8_e_-4_
0x7FU32
0o2F64
0b12
