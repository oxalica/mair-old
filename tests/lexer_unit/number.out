Ok([(Literal(IntLike { ty: None, val: 1 }), "1")])
Ok([(Literal(IntLike { ty: Some(Apply(Angle { name: Path { is_absolute: false, comps: [Name { name: Ok("i32"), hint: None }] }, args: [] })), val: 1 }), "1i32")])
Ok([(Literal(FloatLike { ty: Some(Apply(Angle { name: Path { is_absolute: false, comps: [Name { name: Ok("f64"), hint: None }] }, args: [] })), val: 1200 }), "1.2e3f64")])
Ok([(Literal(FloatLike { ty: None, val: 0.0012 }), "1.2e-3")])
Ok([(Literal(FloatLike { ty: None, val: 1000 }), "1e+3")])
Ok([(Literal(IntLike { ty: None, val: 61284 }), "0xEf64")])
Ok([(Literal(FloatLike { ty: Some(Apply(Angle { name: Path { is_absolute: false, comps: [Name { name: Ok("f64"), hint: None }] }, args: [] })), val: 71 }), "0o__1_07f64")])
Ok([(Literal(IntLike { ty: None, val: 5 }), "0b__1_01")])
Ok([(Literal(FloatLike { ty: None, val: 132800 }), "1__3_.2_8_e_4_")])
Ok([(Literal(FloatLike { ty: None, val: 0.001328 }), "1__3_.2_8_e-_4_")])
Err(LexicalError { loc: "1__3_.2_8_e_-4_", kind: InvalidNumberSuffix })
Err(LexicalError { loc: "0x7FU32", kind: InvalidNumberSuffix })
Err(LexicalError { loc: "0o2F64", kind: InvalidNumberSuffix })
Err(LexicalError { loc: "0b12", kind: InvalidNumberSuffix })
